A deprecation warning when openai version < 1.0

### Context

Python version: 3.11.4
The current version of `pandasai` being used: 1.5.4

___

### 🐛 Describe the bug

### Short description

There is a recently merged update [containing upgrade to OpenAI v1](https://github.com/gventuri/pandas-ai/pull/753/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R15).
Therefore, there is some problems running a newest version of pandasai in the same old environment (without upgrading `openai v0.27`>`openai ^1.0`).

___

### Details

Consider this issue more like a topic to discuss: should we add in [`is_open_ai_v1()`](https://github.com/gventuri/pandas-ai/blob/8093e8da58ea6090013eb5a8a73c49c550f8b189/pandasai/helpers/openai.py#L8) utility function a deprecation warning? Or even exception? 
I guess it'd be a plus. Since otherwise it falls with quite obscured traceback, some part of the logs i put below:
```
Generate python code and return full updated code:
2023-11-28 22:54:02 [INFO] Executing Step 2: CodeGenerator
2023-11-28 22:54:02 [ERROR] Pipeline failed on step 2: 'OpenAI' object has no attribute 'api_base'
```
Expl.: in the above case, if we use `openai` < 1.0, `is_openai_v1()` [obviously returns `False`](https://github.com/gventuri/pandas-ai/blob/8093e8da58ea6090013eb5a8a73c49c550f8b189/pandasai/llm/base.py#L257C16-L257C30), and then it [tries to access `self.api_base`](https://github.com/gventuri/pandas-ai/blob/8093e8da58ea6090013eb5a8a73c49c550f8b189/pandasai/llm/base.py#L261) which has never been exists in this turn so far. 

___

### Steps to reproduce

1. Downgrade openai package. (`$ pip uninstall openai && pip install openai==0.*`)
2. Take the very basic example. Like this, for instance:
```python
import pandas as pd
from data.sample_dataframe import dataframe

from pandasai import SmartDataframe
from pandasai.llm import OpenAI

df = pd.DataFrame(dataframe)

llm = OpenAI()
df = SmartDataframe(df=pd.DataFrame(dataframe), config={"llm": llm})
response = df.chat("Calculate the sum of the gdp of north american countries")
print(response)
```

3. Run code.

--- a/pandasai/llm/base.py
+++ b/pandasai/llm/base.py
@@ -178,7 +178,7 @@ class BaseOpenAI(LLM):
     """
 
     api_token: str
-    api_base: str
+    api_base: str = "https://api.openai.com/v1"
     temperature: float = 0
     max_tokens: int = 1000
     top_p: float = 1

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "pandasai"
-version = "1.5.4"
+version = "1.5.5"
 description = "PandasAI is a Python library that integrates generative artificial intelligence capabilities into Pandas, making dataframes conversational."
 authors = ["Gabriele Venturi"]
 license = "MIT"
@@ -12,7 +12,7 @@ python = ">=3.9,<3.9.7 || >3.9.7,<4.0"
 python-dotenv = "^1.0.0"
 pandas = "1.5.3"
 astor = "^0.8.1"
-openai = "^1.0.0"
+openai = "<2"
 ipython = "^8.13.1"
 matplotlib = "^3.7.1"
 pydantic = "^1"

